### ISSUE TITLE
'minetest.get_version()' function

### ISSUE THREAD

**Issue by runsy** [minetest/minetest#7985]
- Feature request
 

Summary:

It would be cool for a better mod compatibility between different Minetest versions.

**Comment by ClobberXD** [comment]
We've already got `minetest.get_player_info`:
https://github.com/minetest/minetest/blob/24cdbe9bf5fae88d76252add1b4ba0f8a7d48697/doc/lua_api.txt#L3418-L3440

Note that the actual version string (like `"0.4.17.1"`) can only be obtained using debug builds. This string is useless anyway, as not all compatible forks use the same versioning system Minetest does. To ensure mod compatibility, make use of `protocol_version` instead.

**Comment by SmallJoker** [comment]
This problem was usually solved by adding flags to `minetest.features` so that mods can check whether the current Minetest version supports the feature.

**Comment by Wuzzy2** [comment]
RTFM!

```
* `minetest.get_version()`: returns a table containing components of the
   engine version.  Components:
    * `project`: Name of the project, eg, "Minetest"
    * `string`: Simple version, eg, "1.2.3-dev"
    * `hash`: Full git version (only set if available), eg, "1.2.3-dev-01234567-dirty"
  Use this for informational purposes only. The information in the returned
  table does not represent the capabilities of the engine, nor is it
  reliable or verifyable. Compatible forks will have a different name and
  version entirely. To check for the presence of engine features, test
  whether the functions exported by the wanted features exist. For example:
  `if minetest.nodeupdate then ... end`.
```

Downside: Really can only be used as read-only string, you are actively discouraged for detecting compability.

`minetest.features` sounds nice on paper, but the documentation is incomplete and it's also *really* small; I think a lot of features are missing.
